### ISSUE TITLE
Flutter app won't get notified about the quick link event if the app is at foreground on Android devices

### ISSUE THREAD

**Issue by lukeshan-ctm** [RodrigoSMarques/flutter_branch_sdk#193]
Sorry, have to re-open this issue since it is not resolved yet.

Describe the bug
We are developing a Flutter app running on both Android and iOS platforms. The SDK we use is https://pub.dev/packages/flutter_branch_sdk. On iOS, everything works fine. On Android, however the app won't get notified about the quick link event if the app is at foreground. When that happens I can see logs from the Branch SDK saying "onNewIntent call", but the listener in Flutter app side is not called.

To Reproduce

Setup properly the Branch.io in Flutter app.
Launch the Flutter app on Android devices.
When the app is running at foreground, use adb command to send a valid quick link to the app.
```
adb shell am start -a android.intent.action.VIEW \
    -c android.intent.category.BROWSABLE \
    -d "https://xxxxxx.app.link"
```
The Branch listener is not called.
Expected behavior
The Branch listener should be called.

Screenshots

Smartphone (Please complete the following information. remove session if not platform):

Flutter: version: [e.g. 3.3.5]
Package version: [e.g. 6.2.1]
Device: [e.g. Pixel 6a]
OS: [e.g. Android 13]

**Comment by RodrigoSMarques** [comment]
In Branch's [documentation](https://help.branch.io/developers-hub/docs/android-full-reference) there is no reference to using this command for testing.

I did a debug by the example application. The **onNewIntent** method is successfully called, but according to the Branch documentation available [here](https://help.branch.io/developers-hub/docs/android-basic-integration#initialize-branch), the Intent must have an extra parameter: **branch_force_new_session**.

As there is no extra, the Branch SDK is not triggered.

I don't know how to pass this parameter through ADB.

As this may be an issue in the native SDK, I suggest opening a support ticket for Branch:

[https://help.branch.io/using-branch/page/submit-a-ticket](https://help.branch.io/using-branch/page/submit-a-ticket)

**Comment by RodrigoSMarques** [comment]
I found the problem. I'm working on a new version with the fix.

**Comment by RodrigoSMarques** [comment]
Fix in version 6.4.0
